actions.yaml

```diff
@@ -18,7 +18,7 @@ register-user:
       type: string
     admin:
       description: Whether to create an admin user ("yes" or "no")
-      type: boolean
+      type: string
```

Comment:
That's an interesting one. Supplying "yes" or "no" maps to the cli option you can pass when creating users, so it might be good to expose the same option via a juju action. In yaml terms, "yes" and "no" are boolean, like 0 and 1, true and false. I guess that breaks when entering juju land, though I'm not sure? If I change this to type boolean, can I still use "yes" and "no", but then the other options as well?

Reply:
I was thinking about this one and it might not have a reason anymore for being a string since the user is registered via API now and there the parameter is a boolean. You can see a POST request example [here](https://matrix-org.github.io/synapse/v1.41/admin_api/register_api.html). Actually, being a bool would make things easier, not needing to validate anything. I'll change this.

Reply:
I didnt test if is possible to use yes/1/true while passing parameters to an action, its a good question.

Reply:
In general - we don't need to make the charm be consistent with manually configuring synapse. It would be good to think about what an operator would expect - especially one that isn't that familiar with the application

Reply:
I'm closing this one since the parameter is a boolean now but feel free to reopen it or add new reviews.